UGI-216004-main (2)/UGI-216004-main/PhotoEnhancer/PhotoEnhancer/Program.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Security.Cryptography;
+
+namespace PhotoEnhancer
+{
+    internal static class Program
+    {
+        /// <summary>
+        /// Главная точка входа для приложения.
+        /// </summary>
+        [STAThread]
+        static void Main()
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+
+            var mainForm = new MainForm();
+
+            mainForm.AddFilter(new PixelFilter<LighteningParameters>(
+                "Осветление/затемнение",
+                (pixel, parameters) => pixel * parameters.Coefficient));
+            
+            mainForm.AddFilter(new PixelFilter<EmptyParameters>(
+                "Оттенки серого",
+                (pixel, parameters) =>
+                {
+                    var lightness = 0.3 * pixel.R + 0.6 * pixel.G + 0.1 * pixel.B;
+                    return new Pixel(lightness, lightness, lightness);
+                }));
+
+            mainForm.AddFilter(new PixelFilter<NoiseParameters>(
+                "Шум",
+                (pixel, parameters) =>
+                {
+                    Random rnd = new Random();
+                    double rr = rnd.Next(0, 1);
+                    double rg = rnd.Next(0, 1);
+                    double rb = rnd.Next(0, 1);
+                    return new Pixel(parameters.NoiseIntensity * rr + (1 - parameters.NoiseIntensity) * pixel.R,
+                            parameters.NoiseIntensity * rg + (1 - parameters.NoiseIntensity) * pixel.G,
+                            parameters.NoiseIntensity * rb + (1 - parameters.NoiseIntensity) * pixel.B);
+                }));
+
+            mainForm.AddFilter(new TransformFilter(
+                "Отражение по горизонтали",
+                size => size,
+                (point, size) => new Point(size.Width - point.X - 1, point.Y)
+                ));
+
+            mainForm.AddFilter(new TransformFilter(
+                "Поворот на 90° против ч. с.",
+                size => new Size(size.Height, size.Width),
+                (point, size) => new Point(size.Width - point.Y - 1, point.X)
+                ));
+
+            mainForm.AddFilter(new TransformFilter(
+            "Отражение",
```

Comment:
Название фильтра, которое будет отображаться в списке фильтров, ни о чем не говори пользователю, кроме того, что что-там будет отражаться. Лучше было взять название из задания
```suggestion
            "Отражение относительно главной диагонали",
```
Можно было и  так
```suggestion
            "Транспонирование",
```
но этот термин  требует определенных усилий для понимания, если ты не знаком с алгеброй матриц.